### PR TITLE
NO-JIRA: (feat) ingress certificates gather

### PR DESCRIPTION
**Pull request by rluders**
This PR implements a new data enhancement to...

## Categories

- [ ] Bugfix
- [X] Data Enhancement
- [ ] Feature
- [ ] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample Archive

- `path/to/sample_data.json`

## Documentation

- `path/to/documentation.md`

## Unit Tests

- `path/to/file_test.go`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

Yes/No

## References

https://issues.redhat.com/browse/???
https://access.redhat.com/solutions/???

### CODE REVIEW

**Comment by rluders** [comment]
I need to fix the unit tests for this one, but it can be reviewed.

File: pkg/gatherers/clusterconfig/gather_cluster_ingress_certificates.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+const ingressCertificatesLimits = 64
+
+var ingressNamespaces = []string{
+	"openshift-ingress-operator",
```
Comment (tremes) [inline pkg/gatherers/clusterconfig/gather_cluster_ingress_certificates.go:24]: Does it mean that it's possible (supported) to create an Ingress controller also in this namespace please?

File: pkg/gatherers/clusterconfig/gather_cluster_ingress_certificates.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+		for _, controller := range controllers.Items {
+
+			// Step 4: Check the certificate limits
+			if len(certificates) >= ingressCertificatesLimits {
+				klog.V(2).Infof("Reached the limit of ingress certificates (%d), skipping additional certificates", ingressCertificatesLimits)
+				break
+			}
+
+			if controller.Spec.DefaultCertificate != nil {
+				certName := controller.Spec.DefaultCertificate.Name
```
Comment (tremes) [inline pkg/gatherers/clusterconfig/gather_cluster_ingress_certificates.go:100]: nit: `controller.Spec.DefaultCertificate.Name` is not a certificate name, but secret name

File: pkg/gatherers/clusterconfig/gather_cluster_ingress_certificates.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+				certInfo, certErr := getCertificateInfoFromSecret(ctx, coreClient, namespace, certName)
+				if certErr != nil {
+					errs = append(errs, certErr)
+					continue
+				}
+
+				// Step 5: Add certificate info to the certificates list
+				found := false
```
Comment (tremes) [inline pkg/gatherers/clusterconfig/gather_cluster_ingress_certificates.go:108]: Wouldn't be better to use a map for storing the `ControllerInfo` ?

File: pkg/gatherers/clusterconfig/gather_cluster_ingress_certificates.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+	NotAfter    metav1.Time      `json:"not_after"`
+	Controllers []ControllerInfo `json:"controllers"`
+}
+
+type ControllerInfo struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+}
+
+func (g *Gatherer) GatherClusterIngressCertificates(ctx context.Context) ([]record.Record, []error) {
```
Comment (tremes) [inline pkg/gatherers/clusterconfig/gather_cluster_ingress_certificates.go:41]: Documentation is missing here please

**Comment by tremes** [comment]
Reused in https://github.com/openshift/insights-operator/pull/945

**Pull request closed without merging**